File: go/algorithms/warmup/staircase.go
package main

import "fmt"

func main() {
	var n int
	fmt.Scanf("%v\n", &n)
	for i := 0; i < n; i++ {
		var spaces = ""
		for j := 0; j < (n - 1) - i; j++ {
			spaces += " "
		}

		var pounds = ""
		for k := 0; k < i + 1; k++ {
			pounds += "#"
		}

		fmt.Println(spaces + pounds)
	}
}

File: java/src/algorithms/warmup/Staircase.java
package algorithms.warmup;

import java.util.*;

public class Staircase {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = Integer.parseInt(in.nextLine());

        for (int i = 0; i < n; i++) {
            String spaces = "";
            for (int j = 0; j < (n - 1) - i; j++) {
                spaces += " ";
            }

            String pounds = "";
            for (int k = 0; k < i + 1; k++) {
                pounds += "#";
            }

            System.out.println(spaces + pounds);
        }
    }
}
